DuckDuckGo/TabManager.swift

```diff
@@ -224,6 +224,18 @@ class TabManager {
 
     func save() {
         model.save()
+        assertTabPreviewCount()
```

Comment:
The only concern I have is how often this is called (especially when loading the page). This is simple, as 'save' is a single point to catch all model changes - which is very good, but on the other hand not all of the changes result in preview creation/deletion causing potentially unnecessary checks..